Gems/Atom/Tools/MaterialEditor/Code/atommaterialeditor_files.cmake

```diff
@@ -0,0 +1,73 @@
+
+    Source/Viewport/InputController/MaterialEditorViewportInputControllerBus.h
+    Source/Viewport/MaterialViewportSettings.h
+    Source/Viewport/MaterialViewportRequestBus.h
+    Source/Viewport/MaterialViewportNotificationBus.h
+    Source/Viewport/InputController/MaterialEditorViewportInputController.cpp
+    Source/Viewport/InputController/MaterialEditorViewportInputController.h
+    Source/Viewport/InputController/Behavior.cpp
+    Source/Viewport/InputController/Behavior.h
+    Source/Viewport/InputController/DollyCameraBehavior.cpp
+    Source/Viewport/InputController/DollyCameraBehavior.h
+    Source/Viewport/InputController/IdleBehavior.cpp
+    Source/Viewport/InputController/IdleBehavior.h
+    Source/Viewport/InputController/MoveCameraBehavior.cpp
+    Source/Viewport/InputController/MoveCameraBehavior.h
+    Source/Viewport/InputController/PanCameraBehavior.cpp
+    Source/Viewport/InputController/PanCameraBehavior.h
+    Source/Viewport/InputController/OrbitCameraBehavior.cpp
+    Source/Viewport/InputController/OrbitCameraBehavior.h
+    Source/Viewport/InputController/RotateEnvironmentBehavior.cpp
+    Source/Viewport/InputController/RotateEnvironmentBehavior.h
+    Source/Viewport/InputController/RotateModelBehavior.cpp
+    Source/Viewport/InputController/RotateModelBehavior.h
+    Source/Viewport/MaterialViewportSettings.cpp
+    Source/Viewport/MaterialViewportComponent.cpp
+    Source/Viewport/MaterialViewportComponent.h
+    Source/Viewport/MaterialViewportWidget.cpp
+    Source/Viewport/MaterialViewportWidget.h
+    Source/Viewport/MaterialViewportWidget.ui
+
+    Source/Window/MaterialEditorWindowSettings.h
+    Source/Window/MaterialEditorBrowserInteractions.h
+    Source/Window/MaterialEditorBrowserInteractions.cpp
+    Source/Window/MaterialEditorWindow.h
+    Source/Window/MaterialEditorWindow.cpp
+    Source/Window/MaterialEditorWindowSettings.cpp
+    Source/Window/MaterialEditor.qrc
+    Source/Window/MaterialEditor.qss
+    Source/Window/SettingsDialog/SettingsDialog.cpp
+    Source/Window/SettingsDialog/SettingsDialog.h
+    Source/Window/SettingsDialog/SettingsWidget.cpp
+    Source/Window/SettingsDialog/SettingsWidget.h
+    Source/Window/CreateMaterialDialog/CreateMaterialDialog.cpp
+    Source/Window/CreateMaterialDialog/CreateMaterialDialog.h
+    Source/Window/CreateMaterialDialog/CreateMaterialDialog.ui
+    Source/Window/ToolBar/MaterialEditorToolBar.h
+    Source/Window/ToolBar/MaterialEditorToolBar.cpp
+    Source/Window/ToolBar/ModelPresetComboBox.h
+    Source/Window/ToolBar/ModelPresetComboBox.cpp
+    Source/Window/ToolBar/LightingPresetComboBox.h
+    Source/Window/ToolBar/LightingPresetComboBox.cpp
+    Source/Window/MaterialInspector/MaterialInspector.h
+    Source/Window/MaterialInspector/MaterialInspector.cpp
+    Source/Window/ViewportSettingsInspector/ViewportSettingsInspector.h
+    Source/Window/ViewportSettingsInspector/ViewportSettingsInspector.cpp
```

Comment:
AtomMaterialEditor target is reusing all these classes. These are the ones I mentioned in the other comment that it'd be nice to have them in a library. It'd simplify both AtomMaterialEditor and Material editor executable.

Reply:
I'm anticipating that most of the camera related files will go away if changing to Tom's modular camera system. There would not be enough files left to justify a gem and you're creating a library by creating a new module and target.

Maybe something more like
MaterialEditor.Static (library with all of the files except for the application)
MaterialEditor.Editor (module for editor integration)
MaterialEditor (executable)

Reply:
> Maybe something more like
> MaterialEditor.Static (library with all of the files except for the application)
> MaterialEditor.Editor (module for editor integration)
> MaterialEditor (executable)

Correct, this is how I imagined the library separation.

I'm ok leaving them in the same gem for now. If in the future we need standalone MaterialEditor to be more independent we can split it in two like these:

How it'd be with 1 gem:
````
Gem MaterialEditor --> Depends on O3DEMaterialEditor gem
     MaterialEditor.Static 
     MaterialEditor.Editor (editor integration) --> Depends on MaterialEditor.Static lib and O3DEMaterialEditor module
     MaterialEditor (executable)  --> Depends on MaterialEditor.Static lib
````

How it'd be with 2 gems:
````
Gem MaterialEditor
     MaterialEditor.Static 
     MaterialEditor (executable)  --> Depends on MaterialEditor.Static lib

Gem MaterialEditorIntegration  --> Depends on `MaterialEditor` gem and `O3DEMaterialEditor` gem
     MaterialEditor.Editor (editor integration) --> Depends on MaterialEditor.Static lib and O3DEMaterialEditor module
````
Notice how `MaterialEditor` gem is now independent from anything from O3DE.

Notice how `MaterialEditorIntegration` depends on both `MaterialEditor` and `O3DEMaterialEditor` to integrate itself into O3DE, but without interfering into `MaterialEditor` gem.